### PR TITLE
docs: document setDataProvider to setItems

### PR DESCRIPTION
**Pull request by tomivirkki**
Closes https://github.com/vaadin/docs/issues/1252

### CODE REVIEW

File: articles/upgrading/recommended-changes/index.adoc [hunk 1]
```diff
@@ -40,6 +40,8 @@ See how to use the flag in <<{articles}/flow/configuration/properties#, Configur
 - [classname]`ExecutionCanceler` is replaced with [classname]`PendingJavaScriptResult`.
 - [interfacename]`BootstrapListener` is deprecated in favor of [interfacename]`IndexHtmlRequestListener` when using client-side bootstrapping.
 - The [interfacename]`PageConfigurator` interface is deprecated; customize initial page settings instead by overriding [methodname]`AppShellConfigurator::configurePage()`.
+- [methodname]`setDataProvider()` is now deprecated and it is recommended to use the overloaded [methodname]`setItems()` methods.
+<<../../flow/binding-data/data-provider#lazy-data-binding-using-callbacks, Read more>> about the `dataview` API which was origininally link:https://vaadin.com/releases/vaadin-17[introduced] with Vaadin 17.
```
Comment (tarekoraby) [inline articles/upgrading/recommended-changes/index.adoc:44]:
```suggestion
<<../../flow/binding-data/data-provider#lazy-data-binding-using-callbacks, Read more>> about the `dataview` API which was originally link:https://vaadin.com/releases/vaadin-17[introduced] with Vaadin 17.

```

File: articles/upgrading/recommended-changes/index.adoc [hunk 2]
```diff
@@ -40,6 +40,9 @@ See how to use the flag in <<{articles}/flow/configuration/properties#, Configur
 - [classname]`ExecutionCanceler` is replaced with [classname]`PendingJavaScriptResult`.
 - [interfacename]`BootstrapListener` is deprecated in favor of [interfacename]`IndexHtmlRequestListener` when using client-side bootstrapping.
 - The [interfacename]`PageConfigurator` interface is deprecated; customize initial page settings instead by overriding [methodname]`AppShellConfigurator::configurePage()`.
+- [methodname]`setDataProvider()` is now deprecated and it is recommended to use the overloaded [methodname]`setItems()` methods.
```
Comment (tarekoraby) [inline articles/upgrading/recommended-changes/index.adoc:43]:
```suggestion
- [methodname]`Grid.setDataProvider()` is now deprecated and it is recommended to use the overloaded [methodname]`Grid.setItems()` methods.

```

Comment (tomivirkki) [reply articles/upgrading/recommended-changes/index.adoc:43]: It's not just Grid that had this change.

Comment (tomivirkki) [reply articles/upgrading/recommended-changes/index.adoc]: Listed all the components with the API change (Grid, Select, RadioButtonGroup, CheckboxGroup, ListBox, ComboBox)

**Review by tarekoraby — APPROVED** [review]

**Pull request merged**